front/src/modules/simulationResult/components/SpaceTimeChart/SpaceTimeChart.tsx

```diff
@@ -211,9 +216,29 @@ export default function SpaceTimeChart(props: SpaceTimeChartProps) {
     redrawChart();
   }, [resetChart, rotate, selectedTrain, trainSimulations, height]);
 
+  /* redraw the trains if the time scale range has changed from Timeline */
+  useEffect(() => {
+    if (chart && timeScaleDomain && timeScaleDomain.source !== 'SpaceTimeChart') {
+      const currentTimeRange = timeScaleDomain.range;
+      if (currentTimeRange) {
+        chart.x.domain(currentTimeRange);
+        redrawChart();
+      }
+    }
+  }, [timeScaleDomain]);
+
   /* add behaviour on zoom and mousemove/mouseover/wheel on the new chart each time the chart changes */
   useEffect(() => {
-    if (trainSimulations && selectedTrain) {
```

Comment:
We need to keep at least the selectedTrain. Otherwise since the remove of the HOC, there is a "white screen" when deleting the projected train.
We should maybe add a comment to prevent this in the future (and we can remove the cast on the next line)

Reply:
Nice catch, thanks :)

Reply:
Done